### PR TITLE
(node/rucio.ls.lsst.org) deploy rucio.ls.lsst.org 

### PR DESCRIPTION
**Pull request by dtapiacl**
Requirements per ticket information on IT-5509:

- Install the following rpms, xrootd, xrootd-selinux, xrootd-libs, xrootd-client, xrootd-client-libs, xrootd-server-libs, xrootd-server

- modify the /lib/systemd/system/xrootd@.service and /lib/systemd/system/cmsd@.service and change the User and Group inside these two unit files to saluser and it group (current it is user xrootd)

- Change the owner of directory /etc/xrootd, /var/{log, run, spool}/xrootd to saluser (this probably need to be done every time there is an update of the xrootd rpms)

- Allow user bwhite and wyang to do a) sudo su - saluser and b) sudo systemctl restart xrootd@summit. We will add Sajid Ali to this list of users once he has an account

- Place VM in BTS-LHN network. 

- Mount the filesystems KT mentioned.

### CODE REVIEW

File: hieradata/role/rucio.yaml [hunk 1]
```diff
@@ -1,13 +1,6 @@
 ---
 classes:
   - "profile::core::common"
-  - "profile::core::debugutils"
-  - "profile::core::dmz"
+  - "profile::core::nfsclient"
+  - "profile::core::nfsserver"
```
Comment (cbarria) [inline hieradata/role/rucio.yaml]: why using the nfsserver profile without setting an nfs server? i think you only need the nfsclient

File: site/profile/manifests/core/rucio.pp [hunk 2]
```diff
@@ -2,74 +2,97 @@
...
+    ensure          => 'present',
+    name            => 'XRootD Development Branch RPMs for Enterprise Linux $releasever',
+    baseurl         => 'https://xrootd.web.cern.ch/repo/testing/el$releasever/$basearch',
+    gpgkey          => 'https://xrootd.web.cern.ch/repo/RPM-GPG-KEY.txt',
+    metadata_expire => '1d',
+    repo_gpgcheck   => '1',
+    priority        => '30',
+    gpgcheck        => '1',
+    enabled         => '0',
+  }
```
Comment (csilva-cl) [inline site/profile/manifests/core/rucio.pp]: this repo is not necessary if its disable, is there a reason to leave it ?

File: spec/classes/core/rucio_spec.rb [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          ensure: 'present',
+          name: 'XRootD Development Branch RPMs for Enterprise Linux $releasever',
+          baseurl: 'https://xrootd.web.cern.ch/repo/testing/el$releasever/$basearch',
+          gpgkey: 'https://xrootd.web.cern.ch/repo/RPM-GPG-KEY.txt',
+          metadata_expire: '1d',
+          repo_gpgcheck: '1',
+          priority: '30',
+          gpgcheck: '1',
+          enabled: '0',
+        )
```
Comment (csilva-cl) [inline spec/classes/core/rucio_spec.rb]: it may not be needed

File: site/profile/manifests/core/rucio.pp [hunk 4]
```diff
@@ -1,75 +1,70 @@
...
+    mode   => '0644',
+    owner  => 'saluser',
+    group  => 'saluser',
   }
 
-  #  Install Pip3 Packages
-  package { $pip_packages:
-    ensure   => 'present',
-    provider => 'pip3',
+  file { '/etc/xrootd':
```
Comment (cbarria) [inline site/profile/manifests/core/rucio.pp]: all the similar resources can be grouped into an array to DRY(don't repeat yourself) up the code...
Ex:
```file { ['/var/log', '/var/run', '/var/spool', '/etc/xrootd']```

**Review by cbarria — APPROVED** [review]
Looks Great 👍🏻

File: hieradata/node/rucio.ls.lsst.org.yaml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  ens192:
+    content:
+      connection:
+        id: "ens192"
+        uuid: "810adb94-e2c0-4d41-b66f-1175a5c358ba"
+        type: "ethernet"
+        interface-name: "ens192"
+      ethernet: {}
+      ipv4:
+        address1: "139.229.153.160/24,139.229.153.254"
```
Comment (jhoblitt) [inline hieradata/node/rucio.ls.lsst.org.yaml]: Why does this host need a static IP?

File: site/profile/manifests/core/rucio.pp [hunk 6]
```diff
@@ -1,75 +1,70 @@
...
+  }
 
-  #  Define PIP Packages
-  $pip_packages = [
-    'wheel',
-    'cryptography',
-    'rucio',
-  ]
+  # Fetch the xrootd.repo file
+  exec { 'fetch_xrootd_repo':
```
Comment (jhoblitt) [inline site/profile/manifests/core/rucio.pp]: A good rule of thumb is that if `exec` is being used, it probably isn't the correct approach.  The `yumrepo` type should be configure a yum repo.

File: site/profile/manifests/core/rucio.pp [hunk 7]
```diff
@@ -1,75 +1,70 @@
...
+    ]:
+      ensure  => installed,
+      require => Exec['fetch_xrootd_repo'],
+  }
 
-  #  Generate and sign certificate
-  letsencrypt::certonly { $fqdn:
-    plugin      => 'dns-route53',
-    manage_cron => true,
+  file { '/lib/systemd/system/xrootd@.service':
```
Comment (jhoblitt) [inline site/profile/manifests/core/rucio.pp]: Where does the content of these files come from? An rpm?

File: site/profile/manifests/core/rucio.pp [hunk 8]
```diff
@@ -1,75 +1,70 @@
...
+    ensure => file,
+    mode   => '0644',
+    owner  => 'saluser',
```
Comment (jhoblitt) [inline site/profile/manifests/core/rucio.pp:20]: Why is a service unit owned by `saluser`?

File: site/profile/manifests/core/rucio.pp (hunk 4 above)
Comment (jhoblitt) [reply site/profile/manifests/core/rucio.pp]: These repetitive directory resources should be DRY'd up.

File: site/profile/manifests/core/rucio.pp [hunk 9]
```diff
@@ -1,75 +1,70 @@
...
-    'openssl-devel',
-    'systemd-devel',
-    'zlib-devel',
-    'devtoolset-7',
-    'xrootd',
-    'voms',
-  ]
+class profile::core::rucio {
+  # Ensure the GPG key is imported
+  exec { 'import_xrootd_gpg_key':
```
Comment (jhoblitt) [inline site/profile/manifests/core/rucio.pp]: The `yumrepo` resource is able to import the gpg key.

**Review by jhoblitt — CHANGES_REQUESTED** [review]

File: site/profile/manifests/core/rucio.pp (hunk 4 above)
Comment (dtapiacl) [reply site/profile/manifests/core/rucio.pp]: Code was DRY'd  up for better reading.

File: hieradata/node/rucio.ls.lsst.org.yaml (hunk 5 above)
Comment (dtapiacl) [reply hieradata/node/rucio.ls.lsst.org.yaml]: Was no logic behind this, i assumed this host would require a private one. Configuration was updated for DHCP.

**Pull request closed without merging**